Sources/SimpleSalesforceSDK/Services/WebService.swift

```diff
@@ -107,41 +67,148 @@ class WebService {
+                 completionHandler: @escaping ((Data?,Error?) -> Void)) {
+    let bearerAccessToken = "Bearer \(accessToken)"
+    let url = "\(host)/data/v54.0/query/?q="
+    let fetchQuery = query.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)!
+
+    guard let fetchUrl = URL(string: "\(url)\(fetchQuery)") else {return}
+
+    var request = URLRequest(url: fetchUrl)
+    request.httpMethod = "GET"
+    request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+    request.setValue(bearerAccessToken,forHTTPHeaderField: "Authorization")
+
+    guard let expiry = KeychainService.accessTokenExpiryDate, expiry > Date.now else {
+      return
+    }
 
     let task = URLSession.shared.dataTask(with: request, completionHandler: { data, response, error in
       guard let response = response as? HTTPURLResponse else {return}
-
       if (200...299).contains(response.statusCode) {
-        if let error = error {
-          print(error.localizedDescription)
-        } else {
-          guard let data = data else { return }
-          do {
-            let decoder = JSONDecoder()
-            decoder.dateDecodingStrategy = .secondsSince1970
-            let responseData = try decoder.decode(RefreshTokenResponse.self, from: data)
-            
-            KeychainService.setAccessToken(responseData.accessToken)
-            self.interospectAccessToken(host: host, clientId: clientId, clientSecret: clientSecret, accessToken: responseData.accessToken)
-          } catch {
-            print(error.localizedDescription)
-          }
+        completionHandler(data, nil)
+      } else if response.statusCode == 401 {
+        self.refreshAccessToken(host: host,
+                                clientId: clientId,
+                                clientSecret: clientSecret,
+                                refreshToken: refreshToken){ error in
+
         }
+      } else {
+        completionHandler(nil, error)
       }
     })
     task.resume()
   }
+
+  /// Updates the data using sObject Rows resource.
+  /// - Parameters:
+  ///     - host: The Salesforce instance’s endpoint.
+  ///     - accessToken: The access token issued by salesforce.
+  ///     - id: Record id to update record.
+  ///     - objectName: object name to update record.
+  ///     - fieldUpdates: Update record data.
+  ///     - completionHandler: The block returns no value and takes the following parameter:
+  ///         - error: An error object that contains information about a problem, or nil if the request completed successfully.
+  func updateRecord(host: String,
+                    clientId: String,
+                    clientSecret: String,
+                    refreshToken: String,
+                    accessToken: String,
+                    id: String,
+                    objectName: String,
+                    fieldUpdates: [String: Any],
+                    completionHandler: @escaping ((Error?) -> Void))
+  {
```

Comment:
Please follow some consistent formatting.

In line 133, of this same file you kept the `{` in the same line as ending parenthesis, and here it's not.

Also, these long list of params, makes me think we might want to use a struct for the common params, what do you think?

Something like `SFConfig`?